exponentation_by_squaring/exponentation_by_squaring.c

```diff
@@ -0,0 +1,25 @@
+#include<stdio.h>
+
+/* Algorithm to calculate x raised to power k very efficiently
+*  Can be used as a great alternative to normal pow function
+*  base - base of the expression
+*  power - power of expression
+*/
+
+long int exponentation_by_squaring(int base, int power) {
```

Comment:
The base can be non-integer for the case I guess. Also in that case, we should go with `long double` or at least `double` for the return type of the function.